src/server/api/controllers/vessel-reports.controller.js

```diff
@@ -3,13 +3,84 @@
 const HttpError = require("../lib/utils/http-error");
 const knex = require("../../config/db");
 
-//Get vessels reports by vessel id
-const getVesselsReportByVoyageId = async voyage_id => {
-  const vesselReports = await knex.from("vessel_reports").where({ voyage_id });
-  if (vesselReports.length === 0) {
-    return `No voyage exist with that id ${voyage_id}.`;
+//Get vessels reports by voyage_id with limit, offset=0 and order =desc
+const getVesselsReportByVoyageIdWithParams = async (voyage_id, query) => {
+  const voyage = await knex.from("voyages").where({ id: voyage_id });
+
+  if (voyage.length === 0) {
+    throw new HttpError(
+      "Bad request",
+      `No voyage exist with that id ${voyage_id}.`,
+      404
+    );
+  }
+  if (query.orderBy === undefined) {
+    const vesselReports = await knex
+      .from("vessel_reports")
+      .where({ voyage_id })
+      .orderBy("id", "desc")
+      .limit(query.limit)
+      .offset(0);
+
+    if (vesselReports.length === 0) {
+      throw new HttpError(
+        "Bad request",
+        `No vessel-reports exist with that voyage id ${voyage_id}.`,
+        404
+      );
+    }
+    return vesselReports;
+  } else {
+    const vesselReports = await knex
+      .from("vessel_reports")
+      .where({ voyage_id })
+      .orderBy("id", query.orderBy)
+      .limit(50)
+      .offset(0);
+
+    if (vesselReports.length === 0) {
+      throw new HttpError(
+        "Bad request",
+        `No vessel-reports exist with that voyage id ${voyage_id}.`,
+        404
+      );
+    }
+    return vesselReports;
+  }
```

Comment:
You are making it a lot more complicated than it needs to be. 

I would just set default value on all of my query parameters and then do _a single_ knex statement that works no matter what the params are instead of all of the if-else spaghetti 😄 

```
const getVesselsReportByVoyageIdWithParams = async (voyage_id, query) => {
  const { limit = 50, offset = 0, orderBy = "descending" } = query;

  ...

  const vesselReports = await knex
    .from("vessel_reports")
    .where({ voyage_id })
    .orderBy("id", { orderBy })
    .limit({ limit })
    .offset({ offset })
}
```

Reply:
By the way: check this article to read about destructuring with default values: 

https://wesbos.com/destructuring-default-values/

Reply:
resolved with help of Zaki.